src/shogun2-core/src/main/java/de/terrestris/shogun2/service/AbstractSecuredPersistentObjectService.java

```diff
@@ -0,0 +1,243 @@
+package de.terrestris.shogun2.service;
+
+import java.util.Arrays;
+import java.util.HashSet;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+
+import de.terrestris.shogun2.dao.GenericHibernateDao;
+import de.terrestris.shogun2.dao.PermissionCollectionDao;
+import de.terrestris.shogun2.model.SecuredPersistentObject;
+import de.terrestris.shogun2.model.User;
+import de.terrestris.shogun2.model.UserGroup;
+import de.terrestris.shogun2.model.security.Permission;
+import de.terrestris.shogun2.model.security.PermissionCollection;
+
+/**
+ *
+ * @author Nils Bühner
+ * @see AbstractCrudService
+ *
+ */
+public abstract class AbstractSecuredPersistentObjectService<E extends SecuredPersistentObject, D extends GenericHibernateDao<E, Integer>>
+		extends AbstractCrudService<E, D> {
+
+	@Autowired
+	@Qualifier("permissionCollectionService")
+	protected PermissionCollectionService<PermissionCollection, PermissionCollectionDao<PermissionCollection>> permissionCollectionService;
+
+	/**
+	 * Default constructor, which calls the type-constructor
+	 */
+	@SuppressWarnings("unchecked")
+	public AbstractSecuredPersistentObjectService() {
+		this((Class<E>) SecuredPersistentObject.class);
+	}
+
+	/**
+	 * Constructor that sets the concrete entity class for the service.
+	 * Subclasses MUST call this constructor.
+	 */
+	protected AbstractSecuredPersistentObjectService(Class<E> entityClass) {
+		super(entityClass);
+	}
+
+	/**
+	 * This method adds (user) permissions to the passed entity and persists (!)
+	 * the permission collection!
+	 * If no permissions have been set before, they will be created. Otherwise
+	 * the passed permissions will be added to the existing permission
+	 * collection.
+	 *
+	 * @param entity The secured entity
+	 * @param user The user that gets permissions for the entity
+	 * @param permissions The permissions the user gets for the entity
+	 */
+	public void addUserPermissions(E entity, User user, Permission... permissions) {
```

Comment:
Can we probably communicate via the name that the added permissions are persisted? E.g. `addAndSaveUserPermissions`?